javascript/index.js

```diff
@@ -4,37 +4,128 @@
 
 // Out of sync
-  getInstruction("mashedPotatoes", 0, (step1) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 1, (step2) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 2, (step3) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 3, (step4) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 4, (step5) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
-    document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
-  }, (error) => console.log(error));
+// getInstruction("mashedPotatoes", 0, (step1) => {
+//   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
+// }, (error) => console.log(error));
+
+// getInstruction("mashedPotatoes", 1, (step2) => {
+//   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
+// }, (error) => console.log(error));
 
+// getInstruction("mashedPotatoes", 2, (step3) => {
+//   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
+// }, (error) => console.log(error));
 
+// getInstruction("mashedPotatoes", 3, (step4) => {
+//   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
+// }, (error) => console.log(error));
+
+// getInstruction("mashedPotatoes", 4, (step5) => {
+//   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
+//   document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
+// }, (error) => console.log(error));
 
 // Iteration 1 - using callbacks
+getInstruction("mashedPotatoes", 0, (step0) => {
+  document.querySelector("#mashedPotatoes").innerHTML += `<li>${step0}</li>`;
+  getInstruction("mashedPotatoes", 1, (step1) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
+    getInstruction("mashedPotatoes", 2, (step2) => {
+      document.querySelector(
+        "#mashedPotatoes"
+      ).innerHTML += `<li>${step2}</li>`;
+      getInstruction("mashedPotatoes", 3, (step3) => {
+        document.querySelector(
+          "#mashedPotatoes"
+        ).innerHTML += `<li>${step3}</li>`;
+        getInstruction("mashedPotatoes", 4, (step4) => {
+          document.querySelector(
+            "#mashedPotatoes"
+          ).innerHTML += `<li>${step4}</li>`;
+          document
+            .querySelector("#mashedPotatoesImg")
+            .removeAttribute("hidden");
+        });
+      });
+    });
+  });
+});
 // ...
 
 // Iteration 2 - using promises
 // ...
+obtainInstruction("steak", 0)
```

Comment:
this could be done recursively. think about how that might look :) 

<details>
<summary>potential solution: </summary>

```js
function cookRecipe(recipe, step = 0) {
  obtainInstruction(recipe, step)
    .then((result) => {
      document.querySelector("#" + recipe).innerHTML += `<li>${result}</li>`;
      cookRecipe(recipe, step + 1);
    })
    .catch((err) => console.log(err));
}

cookRecipe("steak");
```

</details>